pkg/asset/manifests/azure/cloudproviderconfig.go

```diff
@@ -51,10 +66,10 @@ func (params CloudProviderConfig) JSON() (string, error) {
 		CloudProviderBackoff:         true,
 		CloudProviderBackoffDuration: 6,
 
-		UseInstanceMetadata: true,
+		UseInstanceMetadata: !azureStack,
```

Comment:
I think I would prefer to keep all of the changes specific to Azure Stack in one `if` block. I find that makes it a bit easier to see exactly what the differences are for Azure Stack.
```
config := config{
   // same code we have in master now
}
if params.CloudName == azure.StackCloud {
	config.ac.UseManagedIdentityExtension = false
	config.ac.AADClientID = params.ClientID
	config.ac.AADClientSecret = params.ClientSecret
	config.UseInstanceMetadata = false
	config.LoadBalancerSku = "basic"
}
```